File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> counter;
        for(const auto& num: nums) {
            counter[num] += 1;
        }

        auto comp = [&counter](int key1, int key2) {
            return counter.at(key1) < counter.at(key2);
        };

        priority_queue<int, vector<int>, decltype(comp)> maxHeap(comp);

        for(const auto& [key, value]: counter) {
            maxHeap.push(key);
        }

        vector<int> res;
        for (int i=0;i<k;i++) {
            auto temp = maxHeap.top();
            cout << temp << endl;
            maxHeap.pop();
            res.push_back(temp);
        }
        return res;
    }
};

File: Data Structures & Algorithms/subsets/submission-2.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []

        def builder(index, temp):
            if index >= len(nums):
                res.append(temp.copy())
                return
            
            # include the num
            temp.append(nums[index])
            builder(index + 1, temp)
            temp.pop()

            # dont include the num
            builder(index + 1, temp)



        builder(0, [])
        return res